library/src/com/google/maps/android/geojson/GeoJsonLayer.java

```diff
@@ -0,0 +1,251 @@
+            throws IOException, JSONException {
+        String line;
+        StringBuilder result = new StringBuilder();
+        // Reads from stream
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+        try {
+            // Read each line of the GeoJSON file into a string
+            while ((line = reader.readLine()) != null) {
+                result.append(line);
+            }
+        } finally {
+            reader.close();
+        }
+        // Converts the result string into a JSONObject
+        return new JSONObject(result.toString());
+    }
+
+    /**
+     * Gets a set of all GeoJsonFeature elements stored
+     *
+     * @return set of GeoJsonFeature elements
+     */
+    public Iterable<GeoJsonFeature> getFeatures() {
+        return mRenderer.getFeatures();
+    }
+
+    /**
+     * Gets the GeoJsonFeature with the given ID if it exists
+     *
+     * @param id of GeoJsonFeature object to find
+     * @return GeoJsonFeature object with matching ID or otherwise null
+     */
+    public GeoJsonFeature getFeatureById(String id) {
+        for (GeoJsonFeature feature : getFeatures()) {
+            if (feature.getId() != null && feature.getId().equals(id)) {
+                return feature;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Adds all the GeoJsonFeature objects parsed from the GeoJSON document onto the map
+     */
+    public void addGeoJsonDataToLayer() {
+        for (GeoJsonFeature feature : mParser.getFeatures()) {
+            addFeature(feature);
+        }
+    }
+
+    /**
+     * Adds a GeoJsonFeature to the GeoJsonLayer
+     *
+     * @param feature GeoJsonFeature to add
+     */
+    public void addFeature(GeoJsonFeature feature) {
+        // Set default styles
+        feature.setPointStyle(mDefaultPointStyle);
+        feature.setLineStringStyle(mDefaultLineStringStyle);
+        feature.setPolygonStyle(mDefaultPolygonStyle);
+        mRenderer.addFeature(feature);
+    }
+
+    /**
+     * Removes a GeoJsonFeature from the GeoJsonLayer
+     *
+     * @param feature GeoJsonFeature to remove
+     */
+    public void removeFeature(GeoJsonFeature feature) {
+        mRenderer.removeFeature(feature);
+    }
+
+    /**
+     * Gets the GoogleMap on which the GeoJsonFeature are displayed
+     *
+     * @return map on which the GeoJsonFeature are displayed
+     */
+    public GoogleMap getMap() {
+        return mRenderer.getMap();
+    }
+
+    /**
+     * Renders all stored GeoJsonFeature on the specified GoogleMap
+     *
+     * @param map to render GeoJsonFeature on, if null all GeoJsonFeature objects are cleared from
+     *            the map
+     */
+    public void setMap(GoogleMap map) {
+        mRenderer.setMap(map);
+    }
+
+    /**
+     * Removes all of the stored GeoJsonFeature objects from the map and clears the mFeatures
+     * hashmap
+     */
+    public void removeGeoJsonLayer() {
```

Comment:
I think this could be named better. 'remove layer', to me at least, implies that the layer remains intact but is no longer attached to the map, whereas this (I think) actually empties out the layer completely, clearing out all data. Perhaps something like 'clear()' or 'reset()'.